File: frontend/src/types/neighborhood.ts
export interface Neighborhood {
  id: null;
  name: string;
  city: string;
  type: NeighborhoodType;
}

export enum NeighborhoodType {
  CERRADO = "CERRADO",
  SEMICERRADO = "SEMICERRADO",
  ABIERTO = "ABIERTO",
}

File: backend/ms-properties/src/main/java/pi/ms_properties/domain/Neighborhood.java
package pi.ms_properties.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Neighborhood")
public class Neighborhood {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Enumerated(EnumType.STRING)
    private NeighborhoodType type;

    @Column(name = "city", nullable = false)
    private String city;

    @Column(name = "latitude", nullable = false)
    private Double latitude;

    @Column(name = "longitude", nullable = false)
    private Double longitude;
}